buildkite/data_source_organization.go

```diff
@@ -1,13 +1,85 @@
 package buildkite
 
 import (
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-func dataSourceOrganization() *schema.Resource {
-	resource := resourceOrganizationSettings()
-	return &schema.Resource{
-		ReadContext: resource.ReadContext,
-		Schema:      resource.Schema,
+type organizationDatasourceModel struct {
+	AllowedApiIpAddresses types.List   `tfsdk:"allowed_api_ip_addresses"`
+	ID                    types.String `tfsdk:"id"`
+	UUID                  types.String `tfsdk:"uuid"`
+}
+
+type organizationDatasource struct {
+	client *Client
+}
+
+func newOrganizationDatasource() datasource.DataSource {
+	return &organizationDatasource{}
+}
+
+func (c *organizationDatasource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	c.client = req.ProviderData.(*Client)
+}
+
+func (*organizationDatasource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_organization"
+}
+
+func (o *organizationDatasource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state organizationDatasourceModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	response, err := getOrganization(o.client.genqlient, o.client.organization)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read organization settings",
+			fmt.Sprintf("Unable to read organization: %s", err.Error()),
+		)
+		return
+	}
+
+	state.ID = types.StringValue(response.Organization.Id)
+	state.UUID = types.StringValue(response.Organization.Uuid)
+	ips, diag := types.ListValueFrom(ctx, types.StringType, strings.Split(response.Organization.AllowedApiIpAddresses, " "))
+	state.AllowedApiIpAddresses = ips
+
+	if diag.HasError() {
+		resp.Diagnostics.Append(diag...)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, state)...)
+}
+
+func (*organizationDatasource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Look up organization settings.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"uuid": schema.StringAttribute{
+				Computed: true,
+			},
+			"allowed_api_ip_addresses": schema.ListAttribute{
+				Computed:    true,
+				ElementType: types.StringType,
```

Comment:
I don't think this value is a `Computed` one. It's optional in the resource and so should also be in the datasource. If it's unset then it is set in the UI as an empty string.

The response when unset is:

```json
{
  "data": {
    "organization": {
      "allowedApiIpAddresses": ""
    }
  }
}
```

Reply:
I'm confused though because this is a datasource. It's not about putting any changes up to the API, its about getting data from the API for use somewhere else in terraform.

Technically I think its all computed by default for a datasource right? Except for maybe the ID or some means to actually identify the object in the API. But since this is tied into the providers `slug` configuration, its redundant to specify it here as well.
Perhaps we can just remove `Computed` and not specify any other fields as well?

Reply:
Ahh from the docs on datasource schema: 

```
// Computed indicates whether the provider may return its own value for
// this Attribute or not. Required and Computed cannot both be true. If
// Required and Optional are both false, Computed must be true, and the
// attribute will be considered "read only" for the practitioner, with 
// only the provider able to set its value.                            
```

So its not required or optional. This field should not be set by the user at all. So it has to be marked as computed